pyiceberg/catalog/rest/__init__.py

```diff
@@ -76,6 +76,43 @@
     import pyarrow as pa
 
 
+class HttpMethod(str, Enum):
+    GET = "GET"
+    HEAD = "HEAD"
+    POST = "POST"
+    DELETE = "DELETE"
+
+
+class Endpoint(IcebergBaseModel):
+    model_config = ConfigDict(frozen=True)
+
+    http_method: HttpMethod = Field()
+    path: str = Field()
+
+    @field_validator("path", mode="before")
+    @classmethod
+    def _validate_path(cls, raw_path: str) -> str:
+        if not raw_path:
+            raise ValueError("Invalid path: empty")
+        raw_path = raw_path.strip()
+        if not raw_path:
+            raise ValueError("Invalid path: empty")
+        return raw_path
+
+    def __str__(self) -> str:
+        """Return the string representation of the Endpoint class."""
+        return f"{self.http_method.value} {self.path}"
+
+    @classmethod
+    def from_string(cls, endpoint: str | None) -> "Endpoint":
+        if endpoint is None:
+            raise ValueError("Invalid endpoint (must consist of 'METHOD /path'): None")
+        elements = endpoint.split(None, 1)
```

Comment:
```suggestion
        elements = endpoint.strip().split(None, 1)
```
strip leading/trailing whitespace before split, just in case